src/main/java/DAO/DBConnection.java

```diff
@@ -0,0 +1,38 @@
+package dao;
+
+import dao.Exceptions.DAOException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DBConnection {
+    private static DBConnection instance;
+    private final DBAccess access;
+
+    public DBConnection(DBAccess access) {
+        this.access = access;
+    }
```

Comment:
My suggestion replace it with Thread-safe Singleton pattern
https://refactoring.guru/design-patterns/singleton/java/example#example-2